crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -1206,6 +1207,66 @@ mod test {
         conductor_handle.shutdown().await.unwrap().unwrap();
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn dump_conductor_state() {
+        holochain_trace::test_run();
+        let uuid = Uuid::new_v4();
+        let dna = fake_dna_zomes(
+            &uuid.to_string(),
+            vec![("zomey".into(), TestWasm::Foo.into())],
+        );
+        let agent_pubkey = fake_agent_pubkey_1();
+
+        let _ = RealRibosomeFixturator::new(crate::fixt::curve::Zomes(vec![TestWasm::Foo]))
+            .next()
+            .unwrap();
+        let (_tmpdir, _, conductor_handle, _agent_key) = setup_app_in_new_conductor(
+            "test app".to_string(),
+            Some(agent_pubkey.clone()),
+            vec![(dna.clone(), None)],
+        )
+        .await;
+        let dna_hash = dna.dna_hash();
+
+        // Allow agents time to join
+        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
Sleeps use to be volatile on CI. Please change this to a timeout with a periodic check that the conductor is reachable. Something like
```rust
tokio::time::timeout(Duration::from_secs(2), async {
  loop {
    tokio::time::sleep(Duration::from_millis(100)).await;
    if conductor.state().cells[0].is_joined() { // I don't know off the top of my head how to best check this.
      break;
    }
  }
})
.await
.unwrap();

Reply:
Is this what happened in the failed CI run? CI proving my point? :joy:

Reply:
I think the CI failure is rather due to differences in the state dump if dpki is installed...probably needs a feature-dependent assertion.